Dashboard/app/js/lib/views/data/inspect-data-table-views.js

```diff
@@ -215,9 +215,33 @@ FLOW.inspectDataTableView = FLOW.View.extend({
 FLOW.DataItemView = FLOW.View.extend({
   tagName: 'span',
   deleteSI: function () {
-    var SI;
+    var SI,slKey, sl;
     SI = FLOW.store.find(FLOW.SurveyInstance, this.content.get('keyId'));
     if (SI !== null) {
+    	// check if we also have the data point loaded locally
+    	slKey = SI.get('surveyedLocaleId');
+    	SL = FLOW.store.filter(FLOW.SurveyedLocale,function(item){
```

Comment:
just a note `FLOW.store.find` makes an extra `XHR` request, while filter uses the client cached data